internal/sandbox/dir.go

```diff
@@ -129,6 +129,27 @@ func (d *Dir) LookupOrCreateDirs(components []string) *Dir {
 	return child.LookupOrCreateDirs(remainder)
 }
 
+// LookupOrFail traverses a set of components and returns the directory containing the final
+// component, or nil if not found.
+func (d *Dir) LookupOrFail(components []string) *Dir {
+	if len(components) == 0 {
```

Comment:
Is this still correct for the case where components is initially zero length (so no recursion happens). Should LookupOrFail return nil in that case, instead of returning d?

This function could also do with an example of what a component/components string look like.

Also, why the "OrFail" suffix... ?